### ISSUE TITLE
Routing failure for same name service (different namespace) when using virtualservice and sidecar resource in Istio 1.11.x

### ISSUE THREAD

**Issue by jleetyro** [istio/istio#35127]
### Bug Description

We recently upgrade from Istio 1.10.4 to 1.11.2 and noticed some requests were failing.

After some digging I have managed to find the issue and have a reproducible example https://github.com/brucenunk/istio-bug

We have two services with the same name in different namespaces. These are both headless services and use virtualservices to route to individual workloads. These requests worked previously but after the upgrade resulted in `503` responses.

After dumping the envoy configuration I noticed that the virtualhost for one of the services doesnt have the virtualservice applied and is routed to the headless service hence disappointment.

This also _only_ happens when a sidecar is defined for the workload.

### Version

```prose
kubectl version: v1.20.7

istioctl version is dockerised so you choose but 1.11.x fails.
```

### Additional Information

example: https://github.com/brucenunk/istio-bug

**Comment by howardjohn** [comment]
Great bug report :+1: looking into it

**Comment by howardjohn** [comment]
`master` works, 1.11 doesn't. So sounds like we fixed the bug didn't backport it

**Comment by howardjohn** [comment]
@ramaraochavali @hzxuzhonghu I couldn't find the exact fix/breaking PR, but I am pretty sure this is relatd to the port 80 changes you two were working on

**Comment by howardjohn** [comment]
Found it: https://github.com/istio/istio/pull/33838/files

**Comment by howardjohn** [comment]
Broken in https://github.com/istio/istio/commit/999f5125f2b33121f233292bb61a4c26306bb7aa, fixed in https://github.com/istio/istio/commit/c8ce180e7a0733269813bd6f9ff3f0735d82c5c0. Sending a PR

**Comment by jleetyro** [comment]
Easy fix then 😉. Thanks for looking into it so quickly.

**Comment by howardjohn** [comment]
Fixed in next 1.11 release: https://github.com/istio/istio/pull/35140
